src/is/types.js

```diff
@@ -158,6 +158,15 @@ export function isWindowObject(value) {
     return value != null && typeof value === 'object' && 'setInterval' in value;
 }
 
+/**
+ * Determines whether a given value is a promise
+ * @param {*} value the value to check
+ * @returns {Boolean} `true` if the value is a promise; otherwise, `false`
+ */
+export function isPromise(value) {
+    return typeof value.then === 'function';
```

Comment:
This is not a [good specification](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) for promises. `{ then : function(){} }` is a regular object with a `then` function, but is considered as a promise by your implementation. 
Also, you should use `is.function` for this kind of verification.

Reply:
Thanks for the update.
How about using `toString.call(value) === '[object Promise]';`

Shall i update with this?